Disputes CRDT - PR 0

lukastanisic99 opened the pull request:
## 📋 Description
Currently everything is broken, especially typescript. To make it easy to digest all these changes I'll introduce them in multiple PRs that build on top of each other (target <- PR 0 <- PR 1 .... <- PR N), so whoever is looking at them can digest parts in isolation. This is the first PR in the chain. **NOT** intended to be merged - merging will happen in the end atomically. 

This PR:
Changes forkId to hash, adds timestamp to onChainSlashes, partitions Snapshot, adds participant field to proof

The logic didn't change to accommodate these changes - just structural. Logic and stability will come later. In general Typescript is a huge mess currently. 

## 💪 Related Issues

- Closes #issue-number

## ✅ Checklist

- [ ] My code compiles without errors.
- [ ] I have performed a self-review of my code.
- [ ] I have written tests that cover the new or changed code.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have linked relevant issues and/or created follow-up issues.
- [ ] I have ran compile, build and test, all works for me

## 🔗 Follow-up Issues

List any follow-up tasks or issues that this PR leads to or creates. If none, write "None".


## 📸 Screenshots (if applicable)

If your changes involve UI modifications, please include screenshots or a brief video demonstration.

lukastanisic99 committed:
change forkId to hash, add timestamp to onChainSlashes, partition Snapshot, add participant to proof

daiagi commented on contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol:
```diff
@@ -74,7 +63,7 @@ contract DisputeManagerFacet is StateChannelCommon {
             totalDeposits: totalDeposits,
             totalWithdrawals: totalWithdrawals,
-            forkCnt: disputeData[dispute.channelId].disputeCommitments.length
+            forkId: disputeData[dispute.channelId].disputeCommitments.length
```
this can't be right
the forkid isn't length

lukastanisic99 replied:
Yeah that's changed in new PRs that stabilized it under the new structure

daiagi commented on contracts/V1/StateChannelDiamondProxy/StateSnapshotFacet.sol:
```diff
@@ -28,7 +28,7 @@ contract StateSnapshotFacet is StateChannelCommon {
         ExitChannelBlock[] memory exitChannelBlocks
     ) external onlySelf {
-        require(getDisputeLength(channelId) == getSnapshotForkCnt(channelId), ErrorDisputeProofRequired());
+        require(getDisputeLength(channelId) == getSnapshotforkId(channelId), ErrorDisputeProofRequired());
```
wrong, at least in the PR
assuming fixed in the folowings

lukastanisic99 replied:
yeah we don't have any of the length stuff for the forks anymore - it's all `IDs`

lukastanisic99 committed:
Disputes CRDT - PR 1 (#84)

* Reduce(disputes) and rename stateData -> SnapshotData

* fix maxSlashCount in reduce

* Disputes CRDT - PR 2 (#85)

* Dispute creation/append, more utility/common functions, DisputeConfirmation and type changes, indexed events, some renames

* rename rename createDispute->uploadDispute and restore timestamp in commitment

* Disputes CRDT - PR 3 (#86)

* DisputeOutput generation, upload and audit, challenge, kill dispute

* Disputes CRDT - PR 4 (#88)

* cleanup/refactor, commit and challenge reduce result

* Disputes CRDT - PR 5 (#89)

* Snapshot update and cleanup

* Snapshot timestamp, genesisTimestamp logic

* Disputes CRDT - PR 6 (#90)

* Split fraud proofs (normal and dispute), update solidity compiler, remove forge submodule (just local now - .gitignore), refactor data types, fixes and general refactor

* evidence/kill time, simplification and some comments

* Disputes CRDT - PR 7 (#91)

* JoinChannel logic

* fix type

* Disputes CRDT - PR 8 (#92)

* small cleanup and refactor

* Deposit/Withdraw assets - cleaned up logic and more refactor

* Disputes CRDT - PR 9 (#94)

* Dispute calldata and refactor/cleanup

* First batch of dispute fraud proofs and more refactor - block utils and dispute utils

* move comment up

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>

---------

Co-authored-by: daiagi <daiagi@gmail.com>
